server/tests/src/test/java/org/infinispan/server/persistence/JdbcStringBasedCacheStorePassivation.java

```diff
@@ -0,0 +1,234 @@
+package org.infinispan.server.persistence;
+
+import static org.infinispan.server.security.Common.sync;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertTrue;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+
+import org.infinispan.client.hotrod.Flag;
+import org.infinispan.client.hotrod.RemoteCache;
+import org.infinispan.client.rest.RestClient;
+import org.infinispan.client.rest.RestResponse;
+import org.infinispan.client.rest.configuration.RestClientConfigurationBuilder;
+import org.infinispan.configuration.cache.CacheMode;
+import org.infinispan.server.test.core.category.Persistence;
+import org.infinispan.server.test.core.persistence.Database;
+import org.infinispan.server.test.junit4.InfinispanServerRule;
+import org.infinispan.server.test.junit4.InfinispanServerTestMethodRule;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+/**
+ * Tests String-based jdbc cache store under the following circumstances:
+ * <p>
+ * passivation == false --all cache entries should always be also in the cache store
+ * preload == true --after server restart, all entries should appear in the cache immediately
+ * purge == false --all entries should remain in the cache store after server  * restart
+ * <p>
+ * Other attributes like singleton, shared, fetch-state do not make sense in single node cluster.
+ *
+ */
+@Category(Persistence.class)
+@RunWith(Parameterized.class)
+public class JdbcStringBasedCacheStorePassivation {
+
+    @ClassRule
+    public static InfinispanServerRule SERVERS = PersistenceIT.SERVERS;
+
+    @Rule
+    public InfinispanServerTestMethodRule SERVER_TEST = new InfinispanServerTestMethodRule(SERVERS);
+
+    private final Database database;
+
+    @Parameterized.Parameters(name = "{0}")
+    public static Collection<Object[]> data() {
+        String[] databaseTypes = PersistenceIT.DATABASE.getDatabaseTypes();
+        List<Object[]> params = new ArrayList<>(databaseTypes.length);
+        for (String databaseType : databaseTypes) {
+            params.add(new Object[]{databaseType});
+        }
+        return params;
+    }
+
+    public JdbcStringBasedCacheStorePassivation(String databaseType) {
+        this.database = PersistenceIT.DATABASE.getDatabase(databaseType);
+    }
+
+    @Test(timeout = 600000)
+    public void testFailoverWithPassivationDisabled() throws Exception {
```

Comment:
`testFailover()`

Reply:
Btw we should not have a timeout on any test, testng listener will handle that.